rholang/src/main/scala/coop/rchain/rholang/interpreter/accounting/CostAccountingAlg.scala

```diff
@@ -1,44 +1,43 @@
 package coop.rchain.rholang.interpreter.accounting
 
-import cats.Monad
 import cats.effect.Sync
 import cats.effect.concurrent.Ref
-import coop.rchain.rholang.interpreter.accounting
 import cats.implicits._
+import cats.{FlatMap, Monad}
+import coop.rchain.rholang.interpreter.errors.OutOfPhlogistonsError
 
 trait CostAccountingAlg[F[_]] {
   def charge(cost: Cost): F[Unit]
-  def modify(f: CostAccount => CostAccount): F[Unit]
-  def getCost(): F[CostAccount]
-  def setCost(cost: CostAccount): F[Unit]
+  def charge(cost: CostAccount): F[Unit]
+  def get(): F[CostAccount]
 }
 
 object CostAccountingAlg {
-  def apply[F[_]: Sync](init: CostAccount): F[CostAccountingAlg[F]] = Ref[F].of(init).map { state =>
-    new CostAccountingAlg[F] {
-      override def charge(cost: Cost): F[Unit]                    = state.update(_.charge(cost))
-      override def modify(f: CostAccount => CostAccount): F[Unit] = state.update(f)
-      override def getCost: F[CostAccount]                        = state.get
-      override def setCost(cost: CostAccount): F[Unit]            = state.set(cost)
-    }
-  }
+  def of[F[_]: Sync](init: CostAccount): F[CostAccountingAlg[F]] =
+    Ref[F].of(init).map(ref => new CostAccountingAlgImpl[F](ref))
```

Comment:
Turns out the API change was even easier than I thought! :tada: And the Ref stays inside thx to the (harmless, admittedly) `.get`. Feels almost like cheating :D Good work!